### DISCUSSION TITLE
CLI actions to list windows/tabs?

### DISCUSSION THREAD

**Discussion by oschrenk** [ghostty-org/ghostty#3782] [Feature Requests, Ideas] [answered]
I expected a list-tabs or list-windows command/action to be available to help interact with ghostty.

There are a lot of tab commands but none of them help me to list the available/active one

```
new_tab
previous_tab
next_tab
last_tab
goto_tab
move_tab
toggle_tab_overview
```

Or am I missing something?

I found https://github.com/ghostty-org/ghostty/discussions/2353 but the discussion surrounding an api/text protocol but it's lacking use cases. Maybe this is one?

## Motivation

Philosphically I am trying to move away from tmux and want my terminal to actual to the splitting natively, but practically tmux still wins for me.

Apart from just being used to tmux, I mostly keep using it because of it's scripting and binding capabilities. Eg. I use it to script sketchybar [1]. 

To replace tmux I would need to be able to

1. List (active) tabs
2. Dynamically name tabs
3. Send (arbitrary) commands from a keybinding (or ideally have an event system to fire when tabs/splits are opened/closed).
4. Optionally be able to define ghostty layouts for "sessions"

**Comment by mitchellh** [comment, marked as answer]
Requires #2353

**Comment by janpaepke** [comment]
Late to the party, but I ran into the same gap when trying to drop tmux. While we wait for #2353 to land, I ended up building [haunt](https://github.com/janpaepke/haunt) on top of Ghostty's AppleScript API as a stopgap — it lists all windows and tabs, groups them by working directory, and lets you fuzzy-switch between them (including across Spaces).
